### PR TITLE
search functionality

### PR DESCRIPTION
**Pull request by PrettyPearl2302**
## Description
I implemented search functionality by creating another endpoint '/api/search', allowing for user input for the term functionality. i used this endpoint in the search component on the frontend, making user input the value for the 'term' parameter. 

the search results still need to be formatted to be displayed rightly to the user 

## Milestones
Building search functionality

## Resources
[react documentation](https://react.dev/learn/responding-to-events)
student store repo search component

## Test Plan
Screenshot 2023-07-14 at 6 15 42 PM

### CODE REVIEW

File: timify-api/app.js [hunk 1]
```diff
@@ -60,6 +66,9 @@ export async function searchForTerm(term, page, limitPerPage, filterForGenres) {
   `;
 
   const data = await taddyGraphqlRequest(query, { term, page, limitPerPage, filterForGenres });
+    if (data.errors) {
```
Comment (ushanakg) [inline timify-api/app.js:68]: does this if statement need to be indented?

File: timify-api/app.js [hunk 2]
```diff
@@ -29,8 +30,13 @@ async function taddyGraphqlRequest(query, variables) {
       'X-API-KEY': `${process.env.API_KEY}`,
     }
   });
+  console.log(response.data)
   return response.data;
 }
```
Comment (ushanakg) [inline timify-api/app.js]: I think this bracket is also incorrectly formatted? looks like it belongs to the try catch statement but it's not aligned.

File: timify-api/app.js [hunk 3]
```diff
@@ -29,8 +30,13 @@ async function taddyGraphqlRequest(query, variables) {
       'X-API-KEY': `${process.env.API_KEY}`,
     }
   });
```
Comment (ushanakg) [inline timify-api/app.js]: the formatting for the fields + this end bracket also looks off to me. I'm having a hard time reading the separation of things happening.

File: timify-api/app.js [hunk 4]
```diff
@@ -29,8 +30,13 @@ async function taddyGraphqlRequest(query, variables) {
...
+  console.log(response.data)
```
Comment (ushanakg) [inline timify-api/app.js]: do we want to keep this log statement? If you want it for testing that's fine but at the very end you should probably clean up log statements that print data (we usually want to leave only error logging)

**Review by ushanakg — APPROVED** [review]

File: timify-api/app.js [hunk 5]
```diff
@@ -15,7 +15,8 @@ app.use(morgan('combined'))
 const endpointURL = "https://api.taddy.org"
 
 async function taddyGraphqlRequest(query, variables) {
-  const response = await axios({
+  try {
+    const response = await axios({
```
Comment (ghost) [inline timify-api/app.js:19]: Please format this try catch block and the 'headers' field has extra space between field name and bracket.

**Review by ghost — COMMENTED** [review]
Overall looks good but please do formatting before each commit

**Pull request merged**